server/game/gameSystems/DiscardCardsFromHandSystem.ts

```diff
@@ -63,16 +75,19 @@ export class DiscardCardsFromHandSystem<TContext extends AbilityContext = Abilit
         for (const player of properties.target as Player[]) {
             const availableHand = player.hand.filter((card) => properties.cardCondition(card, context));
 
+            // Select this player if its their own hand, or the active player from the context if its the 'opponent' choosing
+            const discardingPlayer = properties.discardingPlayerType === RelativePlayer.Self ? player : context.player;
+
             Contract.assertNonNegative(derive(properties.amount, player));
 
             const amount = Math.min(availableHand.length, derive(properties.amount, player));
 
-            if (amount === 0) {
+            if (amount === 0 && discardingPlayer.autoSingleTarget) {
```

Comment:
Why does the value of autoSingleTarget matter in this case? Since there are no cards to discard, there's nothing to do in any case right?

Reply:
Yeah I guess the question is -- how will the reveal process work here?  Do we want them prompted and blocked while cards are flipped or just depend on the lookAt system printing?

Reply:
Reverted this back to the original code, nothing to do here for *either* type of discard if there's nothing to viable to discard